components/dom_struct/lib.rs

```diff
@@ -46,6 +46,14 @@ pub fn dom_struct(args: TokenStream, input: TokenStream) -> TokenStream {
                         &self.#ident
                     }
                 }
+
+                impl Eq for #name {}
+
+                impl PartialEq for #name {
+                    fn eq(&self, other: &Self) -> bool {
+                        self.reflector() == other.reflector()
+                    }
```

Comment:
can we just import DomObject here, so other files don't need to import it?

Reply:
also I think it would be more appropriate to do this in derive(DomObject), see https://github.com/servo/servo/blob/main/components/domobject_derive/lib.rs

Reply:
I could not import `DomObject` because it created circular dependencies

Reply:
How? proc_macro does not need actual `DomObject`, it just needs to emit `use crate::DomObject` (here crate actually refers to script crate, that calls this proc macro). This should work:

```suggestion
                    fn eq(&self, other: &Self) -> bool {
                        use crate::DomObject;
                        self.reflector() == other.reflector()
                    }
```

or we could use [fully qualified syntax](https://doc.rust-lang.org/stable/book/ch19-03-advanced-traits.html#fully-qualified-syntax-for-disambiguation-calling-methods-with-the-same-name):
```suggestion
                    fn eq(&self, other: &Self) -> bool {
                        crate::DomObject::reflector(self) == crate::DomObject::reflector(other)
                    }
```

Reply:
Indeed, thanks!